File: src/modules/stored_file/controllers/stored_file_controller.py
from ninja import Router
from django.http import StreamingHttpResponse
from src.shared.decorators.public_decorator import public
from src.modules.stored_file.services.stored_file_download_service import (
    StoredFileDownloadService,
)

router = Router(tags=["Files"])


@router.get("/{uuid}", auth=None)
@public
def download(request, uuid: str):
    file = StoredFileDownloadService.execute(uuid)

    response = StreamingHttpResponse(file["content"], content_type=file["content_type"])
    response["Content-Disposition"] = f'attachment; filename="{file["name"]}"'
    response["Content-Length"] = file["length"]

    return response


@router.get("/public/{uuid}", auth=None)
@public
def download_public(request, uuid: str):
    file = StoredFileDownloadService.execute(uuid, True)

    response = StreamingHttpResponse(file["content"], content_type=file["content_type"])
    response["Content-Disposition"] = f'attachment; filename="{file["name"]}"'
    response["Content-Length"] = file["length"]

    return response

File: src/modules/stored-file/controllers/stored-file.controller.ts
import { Response } from 'express';
import { IRequest } from '../../../shared/domain/interfaces/request.interface';
import { storedFileDownloadService } from '../services/stored-file-download.service';
import { FileAdapter } from '../../../infra/file-system/adapters/file-adapter';
import { generateJsonError } from '../../../shared/utils/error';

class StoredFileController {
  async download(request: IRequest, response: Response) {
    try {
      const uuid: string = request.params.uuid;
      const file: FileAdapter = await storedFileDownloadService.execute(uuid);
      response.set({
        'Content-Type': file.contentType,
        'Content-Length': file.length,
        'Content-Disposition': `attachment; filename="${file.name}"`,
      });
      response.status(200).send(file.content);
    } catch (error) {
      generateJsonError(error, response);
    }
  }

  async downloadPublic(request: IRequest, response: Response) {
    try {
      const uuid: string = request.params.uuid;
      const file: FileAdapter = await storedFileDownloadService.execute(
        uuid,
        true,
      );
      response.set({
        'Content-Type': file.contentType,
        'Content-Length': file.length,
        'Content-Disposition': `attachment; filename="${file.name}"`,
      });
      response.status(200).send(file.content);
    } catch (error) {
      generateJsonError(error, response);
    }
  }
}

export const storedFileController = new StoredFileController();
